src/ast/passes/codegen_llvm.cpp

```diff
@@ -2245,6 +2245,32 @@ AllocaInst *CodegenLLVM::createTuple(
   return buf;
 }
 
+void CodegenLLVM::createTupleCopy(const SizedType &expr_type,
+                                  const SizedType &var_type,
+                                  Value *dst_val,
+                                  Value *src_val)
+{
```

Comment:
Perhaps add `assert(expr_type.IsTupleTy() && var_type.IsTupleTy())` to the beginning of this method to make sure that all the below getters succeed?